packages/core/src/analytics.ts

```diff
@@ -651,12 +651,18 @@ export class SegmentClient {
     this.appState = nextAppState;
   }
 
-  reset() {
+  reset(resetAnonymousId: Boolean = true) {
```

Comment:
Uppercase `Boolean` in TS is an object type. It should be `boolean` to be the primitive.